### DISCUSSION TITLE
Roadmap to testbench for SIM65

### DISCUSSION THREAD

**Discussion by carlo-bramini** [cc65/cc65#2449] [Ideas]
With reference to PR #2334, I was going to think on the additional feature for supporting a testbench for SIM65 CPU emulator:

![immagine](https://github.com/cc65/cc65/assets/30959007/e5fcc87c-bb50-4bf4-b79b-96fd3a0cb1e2)
https://github.com/cc65/wiki/wiki/Before-the-next-release

So, the first question that came in my mind is:

How should this additional feature be organized?
Does the it need to be saved into a separate repository, like people of VICE emulator did for their testbench, or it will be included into the main sources of cc65, just under a subdirectory?

Thank you.

**Comment by mrdudz** [comment]
IF a separate testbench is created, it should live in the "test" directory just like the other tests.

That said - i am not sure if that makes sense. To me it seems more useful to add support to sim65 to the vice testbench directly
